src/UserModule.cc

```diff
@@ -37,11 +37,15 @@
 
 std::deque<std::string> UserModule::module_stack;
 
+static void __attribute__ ((noinline)) print_err(std::string name, const Location &loc,const Context *ctx){
+	std::string locs = loc.toRelativeString(ctx->documentPath());
```

Comment:
If your goal is to reduce stack space, could just just pass this directly to PRINTB?

Reply:
I do not worry about stack usage and runtime during error handling.
My concern is only stack usage during normal runtime.

Having PRINTB in a method that stacks (pun intended), lets the compiler reserve stack space, even if the execution path for the PRINTB is never used. Having the same PRINTB in a noinline methode means, that we only have to use the stack space when the error message is build.

Even if I would be concerned about stack usage when this method is called (during error handling): It is often surprising what the compiler can optimize, so I would look into that before premature code tweaking.
That is the reason for the noinline in the first place is the automatic optimization - the automatic optimization of inlineing a method that is used once makes both the program smaller and faster at runtime - at a small stack penalty, that for this very specific use case is relevant. gcc is really amazing.